File: src/main/java/io/github/oleksivio/telegram/bot/api/model/method/group/ExportChatInviteLink.java
package io.github.oleksivio.telegram.bot.api.model.method.group;

import io.github.oleksivio.telegram.bot.api.model.NetworkError;
import io.github.oleksivio.telegram.bot.core.controller.network.ActionNetworker;
import io.github.oleksivio.telegram.bot.core.model.responses.StringResponse;
import io.github.oleksivio.telegram.bot.core.model.method.ChatAction;
import io.github.oleksivio.telegram.bot.core.model.responses.CommonResponse;

/**
 * @see <a href="https://core.telegram.org/bots/api#exportchatinvitelink">exportChatInviteLink</a>
 */
public class ExportChatInviteLink extends ChatAction<String> {
    private static final String METHOD = "exportChatInviteLink";

    public ExportChatInviteLink(ActionNetworker actionNetworker) {
        super(METHOD, actionNetworker);
    }

    @Override
    protected Class<? extends CommonResponse<String>> getResultWrapperClass() {
        return StringResponse.class;
    }

    @Override
    public ExportChatInviteLink setChatId(Long chatId) {
        pSetChatId(chatId);
        return this;
    }

    @Override
    public ExportChatInviteLink setNetworkErrorListener(NetworkError onNetworkError) {
        pSetNetworkErrorListener(onNetworkError);
        return this;
    }

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/model/method/group/ExportChatInviteLink.kt
package io.github.oleksivio.telegram.bot.api.model.method.group

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.telegram.bot.core.model.ApiDict
import io.github.oleksivio.telegram.bot.core.model.StringResponse
import io.github.oleksivio.telegram.bot.core.model.method.common.ChatAction

/**
 * @see [exportChatInviteLink](https://core.telegram.org/bots/api/#exportchatinvitelink)
 */
class ExportChatInviteLink(
        /**
         * chat_id Integer or String Yes Unique identifier for the target chat or username of the target channel
         */
        @JsonProperty(ApiDict.CHAT_ID_KEY)
        override val chatId: Long
) : ChatAction<String>() {

    override val resultWrapperClass = StringResponse::class

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "exportChatInviteLink"

}
